### PR TITLE
Prepare for publishing

### PR DESCRIPTION
**Pull request by sndrs**
## What does this do?

make this a publishable, barebones package we can start to add things to. attempts to be secure but light-touch (hopefully!), and borrow heavily from things i found useful in https://github.com/guardian/commercial-core

- builds for ES2020
- updated eslint, ts, prettier configs
- added editorconfig, dependabot, codeowners, CI, CD to NPM (once token has been added to repo secrets)
- bumped node version
- updated vscode configs
- added [semantic release](https://semantic-release.gitbook.io/semantic-release/) for CD, and [commitzen](https://github.com/commitizen/cz-cli) and [commitlint](https://commitlint.js.org) to ensure commit messages adhere to [conventional commits](https://www.conventionalcommits.org/)

it's a bit hard to compare main, might be easier to checkout the branch and have a look.

## Why?

I keep thinking of things it might be useful to put in here, so I wanted to make it as easier as possible to do that so we an use it

### CODE REVIEW

**Comment by gtrufitt** [comment]
👁️ 👁️ Nice! Will look closer next week

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+        uses: bahmutov/npm-install@v1
+      - name: Fetch build
+        uses: actions/download-artifact@v2
+        with:
+          name: dist
+          path: dist
+      - name: Release
+        env:
+          GITHUB_TOKEN: ${{ secrets.RELEASE_TOKEN }}
+          NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
```
Comment (gtrufitt) [inline .github/workflows/CI.yml:122]: How does this release? Underneath guardian-developers or something?

File: README.md [hunk 2]
```diff
@@ -1,3 +1,61 @@
 # Libs
 
-A collection of shared libraries for use cross project at The Guardian
+> A collection of shared JavaScript libraries for use in Guardian projects
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
```
Comment (gtrufitt) [inline README.md:6]: Does this need to happen or is it now in CI?

**Review by gtrufitt — APPROVED** [review]
Certainly feels like a set of sensible defaults, nice one!

File: .github/workflows/CI.yml (hunk 1 above)
Comment (sndrs) [reply .github/workflows/CI.yml:122]: yeah exactly

File: README.md (hunk 2 above)
Comment (sndrs) [reply README.md:6]: yeah happens in https://github.com/guardian/libs/blob/de953f00df5d44c205e5c80913896d81242fdabe/.github/workflows/CI.yml#L7-L16

**Pull request merged**